tez-api/src/main/java/org/apache/tez/dag/api/TezConfiguration.java

```diff
@@ -1828,6 +1828,18 @@ public TezConfiguration(boolean loadDefaults) {
       TEZ_PREFIX + "dag.recovery.enabled";
   public static final boolean DAG_RECOVERY_ENABLED_DEFAULT = true;
 
+
+  /**
+   * Boolean value. When set, this enables AM to fail when DAG recovery is enabled and
+   * restarted app master did not find anything to recover
+   * Expert level setting.
+   */
+  @ConfigurationScope(Scope.AM)
+  @ConfigurationProperty(type="boolean")
+  public static final String TEZ_AM_FAILURE_ON_MISSING_RECOVERY =
```

Comment:
nit: can we have a better config name? TEZ_AM_FAILURE_ON_MISSING_RECOVERY_DATA

Reply:
ack, done